tests/specs/api.e2e.js

```diff
@@ -99,4 +102,205 @@ describe('newrelic api', () => {
+      const queryRi = JSON.parse(errorsResult.request.query.ri)
+      expect(twoHundredCharacterString.length).toBe(200)
+      expect(queryRi).toEqual(ri)
+    })
+
+    withBrowsersMatching(reliableUnload)('does not set ri query param if release() is not called', async () => {
+      await Promise.all([
+        browser.testHandle.expectRum(),
+        browser.url(await browser.testHandle.assetURL('api/no-release.html', releaseApiConfig))
+          .then(() => browser.waitForAgentLoad())
+      ])
+
+      const [errorsResult] = await Promise.all([
+        browser.testHandle.expectErrors(),
+        browser.url(await browser.testHandle.assetURL('/'))
+      ])
+
+      expect(errorsResult.request.query).not.toHaveProperty('ri')
+    })
+  })
+
+  describe('setCustomAttribute()', () => {
+    withBrowsersMatching(notIE)('persists attribute onto subsequent page loads until unset', async () => {
+      const [rumResult] = await Promise.all([
+        browser.testHandle.expectRum(),
+        browser.url(await browser.testHandle.assetURL('instrumented.html', { scriptString: 'newrelic.setCustomAttribute(\'testing\', 123, true);' }))
+          .then(() => browser.waitForAgentLoad())
+      ])
+
+      expect(rumResult.request.body.ja).toEqual({ testing: 123 }) // initial page load has custom attribute
+
+      const [rumResultAfterRefresh] = await Promise.all([
+        browser.testHandle.expectRum(),
+        browser.refresh()
+      ])
+
+      expect(rumResultAfterRefresh.request.body.ja).toEqual({ testing: 123 }) // 2nd page load still has custom attribute from storage
+
+      await browser.execute(function () {
+        newrelic.setCustomAttribute('testing', null)
+      })
+
+      const [rumResultAfterUnset] = await Promise.all([
+        browser.testHandle.expectRum(),
+        browser.url(await browser.testHandle.assetURL('instrumented.html'))
+          .then(() => browser.waitForAgentLoad())
+      ])
+
+      expect(rumResultAfterUnset.request.body).not.toHaveProperty('ja') // 3rd page load does not retain custom attribute after unsetting (set to null)
+    })
+  })
+
+  describe('setUserId()', () => {
+    const ERRORS_INBOX_UID = 'enduser.id' // this key should not be changed without consulting EI team on the data flow
+
+    withBrowsersMatching(reliableUnload, notIE)('adds correct (persisted) attribute to payloads', async () => {
+      await Promise.all([
+        browser.testHandle.expectRum(),
+        browser.url(await browser.testHandle.assetURL('instrumented.html', {
+          init: {
+            jserrors: { harvestTimeSeconds: 2 },
+            privacy: { cookies_enabled: true }
+          },
+          scriptString: `
+            newrelic.setUserId(456)
+            newrelic.setUserId({'foo':'bar'})
+            newrelic.noticeError('fake1')
+            `
+        }))
+          .then(() => browser.waitForAgentLoad())
+      ])
+
+      const firstErrorsResult = await browser.testHandle.expectErrors(3000)
```

Comment:
Should probably move this into the `Promise.all` above. Possible edge case here where the page loads / errors harvest before the test runner can fully execute this line of code. Would also remove the timeout value and remove the harvestTimeSeconds above.